### PR TITLE
Dani homework 5

### PR DESCRIPTION
**Pull request by vargad**

### CODE REVIEW

File: course5/homework05/vector.h [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+class Radian {
+    double mRadian;
+public:
+    explicit Radian(double radian) : mRadian(radian) {}
+    Radian(const Degree& degree);
+
+    double Get() const { return mRadian; }
+};
+
+class Degree {
```
Comment (maciekgajewski) [inline course5/homework05/vector.h]: Why don't you just define one class, called 'Angle'?

File: course5/homework05/vector.h [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+{
+public:
+    explicit PolarHelper(double r)
+        : mR(r)
+    {}
+
+    double Get() { return mR; }
+private:
+    double mR;
+};
```
Comment (maciekgajewski) [inline course5/homework05/vector.h:34]: Ans another class called "Length"

File: course5/homework05/vector.h [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    double GetR() const { return std::sqrt(mX*mX + mY*mY); }
+    double GetO() const { return std::atan2(mY, mX); }
+
+    void SetX(double x) { mX = x; }
+    void SetY(double y) { mY = y; }
+private:
+    double mX, mY;
+};
+
+Vector2D operator+(PolarHelper r, Radian o);
```
Comment (maciekgajewski) [inline course5/homework05/vector.h]: I think that operator* makes more sense here than operator+. Just a gut feeling.

File: course5/homework05/vector.h [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    Vector2D(double x, double y);
+    Vector2D(Vector2D const &) = default;
+    Vector2D(Vector2D &&) = default;
+    Vector2D& operator=(Vector2D const &) = delete;
+    Vector2D& operator=(Vector2D &&rhs) = default;
+
+    // vector small enough that it likely better to take as value
+    bool operator==(Vector2D rhs);
+    bool operator!=(Vector2D rhs);
+    Vector2D operator-();
```
Comment (maciekgajewski) [inline course5/homework05/vector.h:48]: Unary minus operator is missing

**Review by maciekgajewski — CHANGES_REQUESTED** [review]
Good, but unary minus is missing

File: course5/homework05/vector.h (hunk 4 above)
Comment (vargad) [reply course5/homework05/vector.h:48]: What you mean on unary minus? I think it's right there.

Comment (maciekgajewski) [reply course5/homework05/vector.h:48]: ah yes. but it isn't tested.
BTW: you could automate the tests

**Pull request closed without merging**